custom_components/frigate/media_source.py

```diff
@@ -437,13 +437,35 @@ def from_str(
         if len(parts) < 2 or parts[1] != cls.get_identifier_type():
             return None
 
+        print(f"The parts are {parts}")
+        tz = "UTC" if (len(parts) == 6 and str(parts[2]).count("-") == 2) else "LOCAL"
+
+        # TODO probably a better way to do this using Frigate APIs with recordings info
         try:
+            if tz == "LOCAL":
+                return cls(
+                    frigate_instance_id=parts[0],
+                    year_month=cls._get_index(parts, 2),
+                    day=cls._get_index(parts, 3),
+                    hour=cls._get_index(parts, 4),
+                    camera=cls._get_index(parts, 5),
+                )
+
+            ymd_parts = str(cls._get_index(parts, 2)).split("-")
+            hour = cls._get_index(parts, 3)
+            time = dt.datetime(
```

Comment:
Nit: `time` is a commonly used builtin Python module, suggest something else for readability.